ksql-engine/src/main/java/io/confluent/ksql/function/UdfCompiler.java

```diff
@@ -126,33 +141,23 @@ private static IScriptEvaluator createScriptEvaluator(final Method method,
     return scriptEvaluator;
   }
 
+
   private static String typeConversionCode(final String type, final int index) {
     if (type.equals("Map") || type.equals("List")) {
       return type + " arg" + index + " = (" + type + ")args[" + index + "];\n";
     }
-    final String argArrayVal = "args[" + index + "]";
-    final String argVarAssignment = "arg" + index + " = ";
+
```

Comment:
`numericValue` can just be in the code template, and then you can simplify this code:

```
private static final String INTEGER_NUMBER_TEMPLATE =
      "else if (args[#INDEX] instanceof Number) arg#INDEX = "
          + "((Number)args[#INDEX]).intValue();\n";

private static final String NUMBER_TEMPLATE =
      "else if (args[#INDEX] instanceof Number) arg#INDEX = "
      + "((Number)args[#INDEX]).#LC_TYPEValue();\n";

private static String typeConversionCode(final String type, final int index) {
    if (type.equals("Map") || type.equals("List")) {
      return type + " arg" + index + " = (" + type + ")args[" + index + "];\n";
    }

    final StringBuilder builder = new StringBuilder();
    builder.append(genericTemplate);
    if (type.equals("Integer")) {
      builder.append(INTEGER_NUMBER_TEMPLATE);
    } else if (!type.equals("String") && !type.equals("Boolean")) {
      builder.append(GENERIC_NUMBER_TEMPLATE);
    }
    builder.append(THROWS_TEMPLATE);

    return builder.toString()
        .replaceAll("#TYPE", type)
        .replaceAll("#LC_TYPE", type.toLowerCase())
        .replaceAll("#INDEX", String.valueOf(index));
  }
```

Reply:
However, I'm still wondering if there is scope for this function to generate invalid code.  Even if we _know_ today that its only going to be called with a known set of `type` parameter values, I think it would still be sensible to validate `type` against a supported list and throw if not supported. This is better then generating code that won't compile.

Reply:
See line 113

Reply:
i don't really see your code as simplification!

Reply:
Well, it switching the code to only have one set of `replaceAll`s and moves the code generation:

```
final String numericValue = type.equals("Integer") ? "intValue()" : type.toLowerCase()
         .append(argVarAssignment)	+          + "Value()";
```

Into the easy to grok constants.